src/clientlayers/StreamRequest.jl

```diff
@@ -144,29 +149,31 @@ function readbody!(stream::Stream, res::Response, buf_or_stream)
                     # if it's a BufferStream, the response body was gzip encoded
                     # so using the default write is fastest because it utilizes
                     # readavailable under the hood, for which BufferStream is optimized
-                    write(body, buf_or_stream)
+                    n = write(body, buf_or_stream)
                 elseif buf_or_stream isa Stream
                     # for HTTP.Stream, there's already an optimized read method
                     # that just needs an IOBuffer to write into
-                    readall!(buf_or_stream, body)
+                    n = readall!(buf_or_stream, body)
                 else
                     error("unreachable")
                 end
             else
                 res.body = read(buf_or_stream)
+                n = length(res.body)
             end
         elseif res.body isa Base.GenericIOBuffer && buf_or_stream isa Stream
             # optimization for IOBuffer response_stream to avoid temporary allocations
-            readall!(buf_or_stream, res.body)
+            n = readall!(buf_or_stream, res.body)
         else
-            write(res.body, buf_or_stream)
+            n = write(res.body, buf_or_stream)
```

Comment:
i really like that all these `write*` functions now return the number of bytes written

i think i tried to make this change in the past and for some reason it was difficult (can't remember why now) ...so i just wonder if we do / should have some tests that we're actually computing these correctly (it's really easy to miss a `n = ...` or `n += ...` somewhere, especially when editing the code, and thereby under-count)

Reply:
Yeah, probably a good idea. Filed an issue to follow up: https://github.com/JuliaWeb/HTTP.jl/issues/1037